File: optimize_images.py
#!/usr/bin/env python3
"""
Script per ottimizzare immagini della galleria:
- Crea versioni thumbnail (350px max) per la galleria
- Crea versioni full (1200px max) per il lightbox
- Ridimensiona e comprime le immagini
"""

from PIL import Image
import os
from pathlib import Path

# Configurazione
PORTFOLIO_DIR = Path(__file__).parent
THUMB_SIZE = (350, 350)
FULL_SIZE = (1200, 1200)
QUALITY_THUMB = 85
QUALITY_FULL = 90

def optimize_image(input_path, output_thumb, output_full):
    """Ridimensiona un'immagine in versioni thumb e full."""
    try:
        img = Image.open(input_path)
        # Converte RGBA in RGB se necessario (per JPG)
        if img.mode in ('RGBA', 'LA'):
            rgb_img = Image.new('RGB', img.size, (255, 255, 255))
            rgb_img.paste(img, mask=img.split()[-1] if img.mode == 'RGBA' else None)
            img = rgb_img
        
        # Thumbnail (piccolo per galleria)
        img_thumb = img.copy()
        img_thumb.thumbnail(THUMB_SIZE, Image.Resampling.LANCZOS)
        ext = Path(input_path).suffix.lower()
        if ext in ['.jpg', '.jpeg']:
            img_thumb.save(output_thumb, 'JPEG', quality=QUALITY_THUMB, optimize=True)
        else:
            img_thumb.save(output_thumb, quality=QUALITY_THUMB, optimize=True)
        
        # Full (grande per lightbox)
        img_full = img.copy()
        img_full.thumbnail(FULL_SIZE, Image.Resampling.LANCZOS)
        if ext in ['.jpg', '.jpeg']:
            img_full.save(output_full, 'JPEG', quality=QUALITY_FULL, optimize=True)
        else:
            img_full.save(output_full, quality=QUALITY_FULL, optimize=True)
        
        print(f"✅ {Path(input_path).name}")
        return True
    except Exception as e:
        print(f"❌ {Path(input_path).name}: {e}")
        return False

# Elabora immagini BASKET (BRY_*.jpg)
print("\n📸 ELABORO BASKET (BRY)...")
bry_files = sorted(PORTFOLIO_DIR.glob("BRY_*.jpg"))
for f in bry_files:
    thumb = PORTFOLIO_DIR / f"{f.stem}-thumb.jpg"
    full = PORTFOLIO_DIR / f"{f.stem}-full.jpg"
    optimize_image(f, thumb, full)

# Elabora immagini CALCIO (MAT*.jpg)
print("\n⚽ ELABORO CALCIO (MAT)...")
mat_files = sorted(PORTFOLIO_DIR.glob("MAT*.jpg"))
for f in mat_files:
    thumb = PORTFOLIO_DIR / f"{f.stem}-thumb.jpg"
    full = PORTFOLIO_DIR / f"{f.stem}-full.jpg"
    optimize_image(f, thumb, full)

# Elabora immagini EDIT
print("\n🎨 ELABORO EDIT...")
edit_dir = PORTFOLIO_DIR / "edit"
for f in sorted(edit_dir.glob("*")):
    if f.is_file() and f.suffix.lower() in ['.jpg', '.jpeg', '.png']:
        thumb = edit_dir / f"{f.stem}-thumb{f.suffix}"
        full = edit_dir / f"{f.stem}-full{f.suffix}"
        optimize_image(f, thumb, full)

print("\n✨ OTTIMIZZAZIONE COMPLETATA!")
print(f"Cartella: {PORTFOLIO_DIR}")
print("Prossimo passo: aggiornare galleria.html per usare le immagini ottimizzate")

File: optimize_images.js
#!/usr/bin/env node
/**
 * Script per ottimizzare immagini della galleria
 * Crea versioni thumb (350px) e full (1200px)
 * 
 * INSTALLAZIONE:
 * 1. npm install sharp
 * 2. node optimize_images.js
 */

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const THUMB_SIZE = 350;
const FULL_SIZE = 1200;
const QUALITY = 90;

async function optimizeImage(inputPath, thumbPath, fullPath) {
  try {
    const filename = path.basename(inputPath);
    
    // Crea thumbnail
    await sharp(inputPath)
      .resize(THUMB_SIZE, THUMB_SIZE, {
        fit: 'cover',
        position: 'center'
      })
      .jpeg({ quality: QUALITY, progressive: true })
      .toFile(thumbPath);
    
    // Crea versione full
    await sharp(inputPath)
      .resize(FULL_SIZE, FULL_SIZE, {
        fit: 'cover',
        position: 'center',
        withoutEnlargement: true
      })
      .jpeg({ quality: QUALITY, progressive: true })
      .toFile(fullPath);
    
    console.log(`✅ ${filename}`);
    return true;
  } catch (err) {
    console.error(`❌ ${path.basename(inputPath)}: ${err.message}`);
    return false;
  }
}

async function main() {
  console.log('\n🚀 INIZIO OTTIMIZZAZIONE IMMAGINI\n');
  
  const portfolioDir = __dirname;
  let count = 0;
  
  // Elabora BASKET (BRY_*.jpg)
  console.log('📸 ELABORO BASKET (BRY)...');
  const bryFiles = fs.readdirSync(portfolioDir)
    .filter(f => f.startsWith('BRY_') && f.endsWith('.jpg'))
    .sort();
  
  for (const file of bryFiles) {
    const inputPath = path.join(portfolioDir, file);
    const thumbPath = path.join(portfolioDir, `${path.parse(file).name}-thumb.jpg`);
    const fullPath = path.join(portfolioDir, `${path.parse(file).name}-full.jpg`);
    
    if (await optimizeImage(inputPath, thumbPath, fullPath)) {
      count++;
    }
  }
  
  // Elabora CALCIO (MAT*.jpg)
  console.log('\n⚽ ELABORO CALCIO (MAT)...');
  const matFiles = fs.readdirSync(portfolioDir)
    .filter(f => f.startsWith('MAT') && f.endsWith('.jpg'))
    .sort();
  
  for (const file of matFiles) {
    const inputPath = path.join(portfolioDir, file);
    const thumbPath = path.join(portfolioDir, `${path.parse(file).name}-thumb.jpg`);
    const fullPath = path.join(portfolioDir, `${path.parse(file).name}-full.jpg`);
    
    if (await optimizeImage(inputPath, thumbPath, fullPath)) {
      count++;
    }
  }
  
  // Elabora EDIT
  console.log('\n🎨 ELABORO EDIT...');
  const editDir = path.join(portfolioDir, 'edit');
  if (fs.existsSync(editDir)) {
    const editFiles = fs.readdirSync(editDir)
      .filter(f => /\.(jpg|jpeg|png)$/i.test(f))
      .sort();
    
    for (const file of editFiles) {
      const inputPath = path.join(editDir, file);
      const ext = path.parse(file).ext;
      const baseName = path.parse(file).name;
      const thumbPath = path.join(editDir, `${baseName}-thumb${ext}`);
      const fullPath = path.join(editDir, `${baseName}-full${ext}`);
      
      if (await optimizeImage(inputPath, thumbPath, fullPath)) {
        count++;
      }
    }
  }
  
  console.log(`\n✨ COMPLETATO! ${count} immagini elaborate.`);
  console.log('\n📝 PROSSIMI STEP:');
  console.log('1. Commit le nuove immagini: git add . && git commit -m "Add optimized images"');
  console.log('2. Push: git push origin main');
  console.log('3. Aggiorna galleria.html per usare le versioni ottimizzate');
  console.log('4. Il lightbox caricherà automaticamente le versioni -full\n');
}

main().catch(err => {
  console.error('Errore:', err.message);
  console.log('\n⚠️  Se vedi "sharp not found", esegui:');
  console.log('npm install sharp');
  process.exit(1);
});
